servers/src/main/java/tachyon/master/file/journal/InodeDirectoryEntry.java

```diff
@@ -49,4 +51,21 @@ public JournalEntryType getType() {
     parameters.put("childrenIds", mChildrenIds);
     return parameters;
   }
+
+  @Override
+  public int hashCode() {
+    return Objects.hashCode(super.hashCode(), mChildrenIds);
+  }
+
+  @Override
+  public boolean equals(Object object) {
+    if (object instanceof InodeDirectoryEntry) {
+      if (!super.equals(object)) {
+        return false;
+      }
+      InodeDirectoryEntry that = (InodeDirectoryEntry) object;
+      return Objects.equal(this.mChildrenIds, that.mChildrenIds);
```

Comment:
In Tachyon codebase, we don't use `this`. Like in this case, we directly use `mChildrenIds`